Sign Up Screen UI and Logic 🎨

hashimsaffarini opened the pull request:
# 🎟️ **Jira Ticket:** [KAN-4](https://diva-shopping.atlassian.net/browse/KAN-4?atlOrigin=eyJpIjoiMTg5YzM3OWZmYmE4NDhiYzlhZjRkYjFlOGVjYWMzYzkiLCJwIjoiaiJ9)

---

# 👨‍🍳 **What's Cooking in This PR?**

### **New Feature** 🎨: Sign Up Screen UI  
- Added **Sign-Up Screen UI**.
- Included new images for the screen.
- Applied styles to the UI.
<img src="https://github.com/user-attachments/assets/9106f0a6-5de2-478c-bc92-a5d83688dbee" width="300" />

### **Refactoring** ♻️: Code Structure  
- Moved **Password Validation File**.
- Relocated **OrLine Widget**.

### **Styling** 💄: AppStyles Enhancements  
- Updated the `AppStyles` class.

---

# 🚀 **How to Test?**
- Navigate to Sign Up screen

hashimsaffarini committed:
feat(signup) 🎨 – Adding the sign-up screen UI and a new image.
refactor(validation) ♻️ – Moving the password validation file to a new location.
refactor(widget) ♻️ – Moving the OrLine widget.
style(app) 💄 – Adding new styles to the AppStyles class.

omarahmedx14 commented on lib/features/login/ui/login_screen.dart:
```diff
@@ -8,7 +8,7 @@ class LoginScreen extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: loginAppBar(),
+      appBar: authMethodsAppBar('Welcome Back!'),
```
please rename it to AuthMethodsAppBar

hashimsaffarini replied:
But this is a function not a widget

omarahmedx14 replied:
then it's better to start with a verb, buildAuthMethodsAppBar or setupAuthMethodsAppBar

hashimsaffarini committed:
- 🚀 finish UI of verification screen
- ➕ add firebase and configration
- finish logic of sign up screen
- finish logic of login screen
- 🚀 add styles and images

omarahmedx14 commented on lib/features/login/data/repos/login_repos.dart:
```diff
@@ -0,0 +1,41 @@
+      return ApiResult.success(userCredential.user!);
+    } on FirebaseAuthException catch (e) {
+      ErrorHandler errorHandler;
+      if (e.code == 'user-not-found') {
```
please think of a better way than if/elses as it may grows in the future.

omarahmedx14 commented on lib/features/sign_up/data/repos/sign_up_repos.dart:
```diff
@@ -0,0 +1,95 @@
+      return ApiResult.success(response);
+    } on FirebaseAuthException catch (e) {
+      ErrorHandler errorHandler;
+      if (e.code == 'weak-password') {
```
same here

omarahmedx14 replied:
why u making validation in data layer ?

omarahmedx14 commented on lib/features/sign_up/logic/cubit/sign_up_cubit.dart:
```diff
@@ -0,0 +1,117 @@
+  String? _verificationId;
+
+  // Method to decide whether to sign up using email or phone
+  void signUp() async {
```
it's better to make different functions one for email and one for phone

omarahmedx14 commented on lib/features/sign_up/ui/widgets/sign_up_screen_body.dart:
```diff
@@ -1,38 +1,86 @@
-              textStyle: TextStyles.font18RobotoRegular,
-              onPressed: () {},
-              borderRadius: 30,
+    return BlocListener<SignUpCubit, SignUpState>(
```
please put this BlocListener into function/widget ... make UI smaller in general.

hashimsaffarini committed:
- 🎨 edit on repos and api folders in data layer
-🎨 Refactor authMethodsAppBar function to setupAuthMethodsAppBar
- 🎨 edit on DI file becuase i make edits in data layes
- 🎨 edit in navigation when succsfull sign up

hashimsaffarini committed:
Merge branch 'main' into Feature/signUp/KAN-4/UI
